### ISSUE TITLE
Failing to boot with Quickstart instructions [Ubuntu 22.04 Jammy, OpenSSL 3]

### ISSUE THREAD

**Issue by ioolkos** [stritzinger/opcua#6]
Proceeding as per the Quickstart instructions, leads to the following error when running `rebar3 as server shell` or `rebar3 as client shell`.
Maybe it's just me or my system? :)

```
Eshell V13.0.3  (abort with ^G)
1> 2023-02-23T09:45:23.178457+01:00 info: Supervisor: {local,ranch_sup}. Started: id=ranch_server,pid=<0.182.0>.
2023-02-23T09:45:23.180269+01:00 info: Application: ranch. Started at: nonode@nohost.
2023-02-23T09:45:23.180348+01:00 info: Application: xmerl. Started at: nonode@nohost.
2023-02-23T09:45:23.180420+01:00 info: Application: quickrand. Started at: nonode@nohost.
2023-02-23T09:45:23.180478+01:00 info: Application: uuid. Started at: nonode@nohost.
2023-02-23T09:45:23.181738+01:00 info: Starting default keychain process...
2023-02-23T09:45:23.241805+01:00 error: Failed loading keychain entry #{alias => server,cert => {priv,opcua,"certificates/server.pem"},is_required => true,is_trusted => false,key => {priv,opcua,"certificates/server.key"}}: decoding_error
2023-02-23T09:45:23.241904+01:00 error: Supervisor: {local,opcua_sup}. Context: start_error. Reason: missing_required_entry. Offender: id=opcua_keychain_default,pid=undefined.
2023-02-23T09:45:23.242057+01:00 error: crasher: initial call: application_master:init/4, pid: <0.187.0>, registered_name: [], exit: {{{shutdown,{failed_to_start_child,opcua_keychain_default,missing_required_entry}},{opcua,start,[normal,[]]}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}, ancestors: [<0.186.0>], message_queue_len: 1, messages: [{'EXIT',<0.188.0>,normal}], links: [<0.186.0>,<0.44.0>], dictionary: [], trap_exit: true, status: running, heap_size: 376, stack_size: 28, reductions: 160; neighbours:
2023-02-23T09:45:23.242248+01:00 notice: Application: opcua. Exited: {{shutdown,{failed_to_start_child,opcua_keychain_default,missing_required_entry}},{opcua,start,[normal,[]]}}. Type: temporary.
2023-02-23T09:45:23.242306+01:00 notice: Application: uuid. Exited: stopped. Type: temporary.
2023-02-23T09:45:23.242337+01:00 notice: Application: quickrand. Exited: stopped. Type: temporary.
2023-02-23T09:45:23.242373+01:00 notice: Application: xmerl. Exited: stopped. Type: temporary.
2023-02-23T09:45:23.242721+01:00 notice: Application: ranch. Exited: stopped. Type: temporary.
2023-02-23T09:45:23.242862+01:00 info: Supervisor: {local,ranch_sup}. Started: id=ranch_server,pid=<0.197.0>.
2023-02-23T09:45:23.242920+01:00 info: Application: ranch. Started at: nonode@nohost.
2023-02-23T09:45:23.243025+01:00 info: Application: xmerl. Started at: nonode@nohost.
2023-02-23T09:45:23.243122+01:00 info: Application: recon. Started at: nonode@nohost.
2023-02-23T09:45:23.249826+01:00 info: Supervisor: {local,runtime_tools_sup}. Started: id=ttb_autostart,pid=<0.204.0>.
===> Booted ranch
2023-02-23T09:45:23.249899+01:00 info: Application: runtime_tools. Started at: nonode@nohost.
===> Booted xmerl
===> Booted recon
===> Booted runtime_tools
===> Failed to boot opcua for reason {{shutdown,
                                                  {failed_to_start_child,
                                                   opcua_keychain_default,
                                                   missing_required_entry}},
                                                 {opcua,start,[normal,[]]}}
 ```

**Comment by ziopio** [comment]
You did not generate the certificates, you have to run` ./scripts/generate_certificates.sh`
Certificates are optional but not if pointed in the config files, we should probably correct the readme.

**Comment by ziopio** [comment]
If you don't care, just remove the keychain configuration from the config files.

**Comment by ioolkos** [comment]
Thanks for the quick answer!
Note this happens with running `./scripts/generate_certificates.sh` before trying to start. (Maybe I need to adapt something somewhere?)

I'll try to remove the keychain config, thanks.

**Comment by ziopio** [comment]
There is another way, and I am going to change this on master soon.
In the config for keychain you can see the files have `is_required => true`, you can set them to `false` and it will not complain.

Other then this, what do you have under `/priv/certificates`? The script did not do its work ?

**Comment by ioolkos** [comment]
Thanks for the options.
I now think it's a local issue. Works for me on Ubuntu 20.04, but not on 22.04 apparently.

**Comment by ziopio** [comment]
I can reproduce this on ubuntu 22.04.1... we'll investigate

**Comment by ziopio** [comment]
OTP 25.2.3 On ubuntu 22.04.2 Jammy cannot decode the PEM certificate.
Problem seams to be related to openssl version 3 that is installed as default.

**Comment by ziopio** [comment]
openssl 3 by default encodes the key in a different format.
Someone could get the traditional RSA format using` openssl genrsa --traditional ...`
But we probably need to handle this format in code and not rely on different scripts.

**Comment by ioolkos** [comment]
@ziopio Nice; can confirm the fix for Ubuntu 22.04.
